### ISSUE TITLE
Not obeying customAgent CA 

### ISSUE THREAD

**Issue by klinquist** [request/request#1490]
Hi,
It seems that although I have  "ca" file (buffer obj)  in a https customAgent, request is not obeying it and still giving me a "Self signed cert in certificate chain..." error.  I've tried passing it in the globalAgent object as well as to the pool...

My environment is an AWS private subnet with a Squid caching proxy performing transparent SSL Bumps.  I have generated a self-signed CA which is working perfectly with modules that use node's core https library (such as aws-sdk).

When I run:
If I run
`openssl s_client -showcerts -connect www.googleapis.com:443`
I get the error I expect:
`Verify return code: 19 (self signed certificate in certificate chain)`

However, as long as I specify my CA, 
`openssl s_client -showcerts -CAfile /etc/ssl/certs/myCA.pem -connect www.googleapis.com:443`
Open SSL shows
`Verify return code: 0 (ok)`

I'm passing this customAgent on to request:

```js
    var customAgent = new https.Agent({
        ca : require('fs').readFileSync(process.env.SSL_CA_CERT),
    });
```

and still get the self-signed cert error. `NODE_DEBUG=request` seems to show the ca buffer as the request is made, but I still get the cert error...

Thanks!

**Comment by froatsnook** [comment]
I don't have any experience with playing with request and CAs, but how are you passing `customAgent` to `request`?  Have you tried using `agentOptions` as is done in this example from the README?

```
request.get({
    url: 'https://api.some-server.com/',
    agentOptions: {
        ca: fs.readFileSync('ca.cert.pem')
    }
});
```

**Comment by klinquist** [comment]
Nevermind.  When declaring the customAgent, I was using  https.globalAgent.ca  instead of  https.globalAgent.options.ca.

**Comment by mainakae** [comment]
Using agentOptions wouldn't solve the problem for Klinquist, I'm affraid. There's more than one way to set the custom CA: via options.agentOptions.ca or via options.ca. In case you are using an organization proxy, the agentOptions.ca will not work! (but options.ca will). So... there's one of two possibilities here: to fix documentation and remove all the agentOptions.ca stuff, or fix the code so that ca can be set while tunneling from agentOptions aswell.

**Comment by mainakae** [comment]
Update: it's being discussed and ¿solved? in  #1289
